server/data/data-jmap/src/test/java/org/apache/james/jmap/api/filtering/FilteringManagementContract.java

```diff
@@ -133,11 +133,36 @@ default void definingEmptyRuleListShouldRemoveExistingRules(EventStore eventStor
     default void allFieldsAndComparatorShouldWellBeStored(EventStore eventStore) {
         FilteringManagement testee = instantiateFilteringManagement(eventStore);
 
-        Mono.from(testee.defineRulesForUser(USERNAME, RULE_FROM, RULE_RECIPIENT, RULE_SUBJECT, RULE_TO, RULE_1)).block();
-
+        Mono.from(testee.defineRulesForUser(USERNAME, Optional.of(Version.INITIAL), RULE_FROM, RULE_RECIPIENT, RULE_SUBJECT, RULE_TO, RULE_1)).block();
 
         assertThat(Mono.from(testee.listRulesForUser(USERNAME)).block())
             .isEqualTo(new Rules(ImmutableList.of(RULE_FROM, RULE_RECIPIENT, RULE_SUBJECT, RULE_TO, RULE_1), new Version(0)));
     }
 
+    @Test
+    default void definingRulesFirstTimeWithWrongVersionInitialShouldFail(EventStore eventStore) {
+        FilteringManagement testee = instantiateFilteringManagement(eventStore);
+
+        assertThatThrownBy(() -> Mono.from(testee.defineRulesForUser(USERNAME, Optional.of(new Version(-2)), RULE_3, RULE_2, RULE_1)).block())
```

Comment:
IMO `new Version(-2)` should throw an Illegal arguent exception in itsel (cn you add a precondition there?)

Also, use `new Version(1)` here...